lib/manageiq/appliance_console/certificate.rb

```diff
@@ -40,8 +41,11 @@ def initialize(options = {})
     end
 
     def request
+      undo_tracking if complete?
+
       if should_request_key?
         principal.register
+        save_self_signed
```

Comment:
Is this really specific to self signed certs or will it just move whatever is at the `key_filename`/`cert_filename` location?

For example, if I run this after having retrieved the certs from the authority, I wouldn't be saving self-signed certs, right? They would be the real ones. Or am I misunderstanding?

Reply:
Once certs have been successfully retrieved from the authority, `should_request_key?`
will return false consequently avoiding this block and method `save_self_signed` from being invoked.

Reply:
Isn't that not true in the rerun case? When you run `undo_tracking` won't that cause `should_request_key?` to return `true` in which case you will be in this block? Granted, in that case the files will have been deleted by `undo_tracking`, but that brings me back to my original point which is, why save the generated ones off at all?

It seems like you're implementing a lot of logic here which boils down to:
  - delete old certs
  - fetch new certs

Whether they're self signed or not doesn't look like it matters.

Reply:
@carbonin I `sort of agree`. I agree removing the self signed certs, when not yet
tracking, is fine but I think we should only invoke `undo_tracking` when certmonger
is actively tracking cert, that is when `complete?` is true.